src/atomate2/vasp/schemas/magnetism.py

```diff
@@ -0,0 +1,105 @@
+
+from atomate2.common.schemas.magnetism import (
+    MagneticOrderingOutput as MagneticOrderingOutputBase,
+)
+from atomate2.common.schemas.magnetism import (
+    MagneticOrderingRelaxation as MagneticOrderingRelaxationBase,
+)
+from atomate2.common.schemas.magnetism import (
+    MagneticOrderingsDocument as MagneticOrderingsDocumentBase,
+)
+
+if TYPE_CHECKING:
+    from emmet.core.tasks import TaskDoc
+
+
+class MagneticOrderingRelaxation(MagneticOrderingRelaxationBase):
+    """
+    Defines the relaxation output document for a magnetic ordering calculation.
+
+    The construction of this document is implemented here for VASP. See base class for
+    more details.
+    """
+
+    @classmethod
+    def from_task_document(cls, task_document: TaskDoc, uuid: str | None = None):
+        """Construct a MagneticOrderingRelaxation from a task document.
+
+        This does not include the uuid, which must be set separately.
+        """
+        dir_name = task_document.dir_name
+        structure = task_document.structure
+        input_structure = task_document.input.structure
+        if not input_structure.site_properties.get("magmom"):
+            input_structure.add_site_property(  # input struct likely has no magmoms
+                "magmom", task_document.input.parameters["MAGMOM"]
+            )
+        energy = task_document.output.energy
+
+        return cls.from_structures_and_energies(
+            input_structure=input_structure,
+            output_structure=structure,
+            output_energy=energy,
+            uuid=uuid,
+            dir_name=dir_name,
+        )
+
+
+class MagneticOrderingOutput(MagneticOrderingOutputBase):
+    """Defines the static output for a magnetic ordering calculation.
+
+    The construction of this document is implemented here for VASP. See base class for
+    more details.
+    """
+
+    @classmethod
+    def from_task_document(
+        cls,
+        task_document: TaskDoc,
+        uuid: str | None = None,
+        relax_output: MagneticOrderingRelaxation | None = None,
+    ):
+        """Construct a MagneticOrderingOutput from a task document.
+
+        This does not include the uuid, which must be set separately.
+        """
+        dir_name = task_document.dir_name
+        structure = task_document.structure
+        input_structure = task_document.input.structure
+        if not input_structure.site_properties.get("magmom"):
+            input_structure.add_site_property(  # input struct likely has no magmoms
+                "magmom", task_document.input.parameters["MAGMOM"]
+            )
+        energy = task_document.output.energy
+
+        return cls.from_structures_and_energies(
+            input_structure=input_structure,
+            output_structure=structure,
+            output_energy=energy,
+            uuid=uuid,
+            dir_name=dir_name,
+            relax_output=relax_output,
+        )
+
+
+class MagneticOrderingsDocument(MagneticOrderingsDocumentBase):
```

Comment:
Thinking about this a bit more, I'm personally not too keen on these subclasses for the schemas. While the implementation is clean, it ultimately adds a bit more boilerplate to implement if you want to configure the magnetism workflow for another code. The other issue is that if you need the full task document to construct the schemas, this will pull a lot of information from the database adding unnecessary strain.

Ideally, the inputs needed for the schemas would be configured in the workflow itself, e.g., see the code for constructing the elastic schemas here: https://github.com/materialsproject/atomate2/blob/main/src/atomate2/common/jobs/elastic.py#L167

and here: https://github.com/materialsproject/atomate2/blob/9066774f4615f48d091ca80dcbbefa3ba1527f9a/src/atomate2/common/flows/elastic.py#L136-L144

Only the specific information needed to construct the documents are passed and there is no VASP specific code.

Is it possible to do something similar in this workflow? That would enable us to remove the VASP specific schemas, and the also the extra class function for building the schemas.



Only the specific information